pkg/logentry/stages/json.go

```diff
@@ -24,6 +25,7 @@ const (
 // JSONConfig represents a JSON Stage configuration
 type JSONConfig struct {
 	Expressions map[string]string `mapstructure:"expressions"`
+	Source      string            `mapstructure:"source"`
```

Comment:
For consistency, we've been handling optional config strings as a pointer to string